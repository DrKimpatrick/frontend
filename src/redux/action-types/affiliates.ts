export enum AffiliateTypes {
  AddMoreInfo = 'Affiliate/AddMoreInfo',
  Errors = 'Affiliate/Errors',
  Loading = 'Affiliate/Loading'
}

// export interface CompanyOrSchool {
//   _id?: string;
//   name: string;
//   address: string;
//   website: string;
//   accountManagerName: string;
//   email: string;
//   phone: string;
//   userId?: string;
//   createdAt?: string;
//   updatedAt?: string;
// }

interface AddMoreInfo {
  type: typeof AffiliateTypes.AddMoreInfo;
  payload: {
    data: any;
  };
}

interface Errors {
  type: typeof AffiliateTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof AffiliateTypes.Loading;
}

export type CompanyActionTypes = AddMoreInfo | Errors | Loading;
