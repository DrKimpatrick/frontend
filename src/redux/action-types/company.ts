export enum CompanyTypes {
  AddCompany = 'Company/AddCompany',
  GetSpecificCompany = 'Company/GetSpecificCompany',
  UpdateCompany = 'Company/UpdateCompany',
  DeleteCompany = 'Company/DeleteCompany',
  Errors = 'Company/Errors',
  Loading = 'Company/Loading'
}

export interface CompanyOrSchool {
  _id?: string;
  name: string;
  address: string;
  website: string;
  accountManagerName: string;
  email: string;
  phone: string;
  userId?: string;
  createdAt?: string;
  updatedAt?: string;
}

interface AddCompany {
  type: typeof CompanyTypes.AddCompany;
  payload: {
    data: CompanyOrSchool;
  };
}

interface Errors {
  type: typeof CompanyTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof CompanyTypes.Loading;
}

export type CompanyActionTypes = AddCompany | Errors | Loading;
