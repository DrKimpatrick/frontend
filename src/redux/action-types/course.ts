import { CourseOwnerType, Course, UserCourseType } from 'redux/actions/course';
import { User } from 'redux/actions/user';

export enum CourseTypes {
  AddCourse = 'CourseTypes/AddCourse',
  UpdateCourse = 'CourseTypes/UpdateCourse',
  ListPendingCourse = 'CourseTypes/ListPendingCourse',
  ListAcceptedCourse = 'CourseTypes/ListAcceptedCourse',
  ListDeclinedCourse = 'CourseTypes/ListDeclinedCourse',
  Loading = 'CourseTypes/Loading',
  Errors = 'CourseTypes/Errors',
  NumberOfPendingCourse = 'CourseTypes/NumberOfPendingCourse',
  NumberOfDeclinedCourse = 'CourseTypes/NumberOfDeclinedCourse',
  NumberOfAcceptedCourse = 'CourseTypes/NumberOfAcceptedCourse',
  ListCourseDetail = 'CourseTypes/ListCourseDetail',
  CourseDetailLoading = 'CourseTypes/CourseDetailLoading',
  ListCourse = 'CourseTypes/ListCourse',
  ListCourseLoading = 'CourseTypes/ListCourseLoading',
  SubmitLoading = 'CourseTypes/SubmitLoading',
  ListCourseOwner = 'CourseTypes/ListCourseOwner',
  CourseOwnerLoading = 'CourseTypes/CourseOwnerLoading',
  UserCourse = 'CourseTypes/UserCourse',
  UserCourseLoading = 'CourseTypes/UserCourseLoading',
  UserWhoPaidCourse = 'CourseTypes/UserWhoPaidCourse',
  UserWhoPaidCourseLoading = 'CourseTypes/UserWhoPaidCourseLoading'
}

export enum CourseStatus {
  Accepted = 'accepted',
  Pending = 'pending',
  Declined = 'declined'
}

export enum CourseTimeFormat {
  Minute = 'Min',
  Hour = 'Hr'
}

export enum CourseLevel {
  Advanced = 'advanced',
  Beginner = 'beginner',
  Intermediate = 'intermediate'
}

export enum CourseBillingOptions {
  ONE_TIME = 'one-time',
  DAILY = 'day',
  WEEKLY = 'week',
  MONTHLY = 'month',
  YEARLY = 'year'
}

export interface CoursePagination {
  offset?: number;
  limit?: number;
}
interface ListAcceptedCourse {
  type: typeof CourseTypes.ListAcceptedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface ListPendingCourse {
  type: typeof CourseTypes.ListPendingCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface ListDeclinedCourse {
  type: typeof CourseTypes.ListDeclinedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}
interface Loading {
  type: typeof CourseTypes.Loading;
  payload: {
    loading: boolean;
  };
}

interface Errors {
  type: typeof CourseTypes.Errors;
  payload: {
    errors: any;
  };
}

interface NumberOfPendingCourse {
  type: typeof CourseTypes.NumberOfPendingCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface NumberOfDeclinedCourse {
  type: typeof CourseTypes.NumberOfDeclinedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface NumberOfAcceptedCourse {
  type: typeof CourseTypes.NumberOfAcceptedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface ListCourseDetail {
  type: typeof CourseTypes.ListCourseDetail;
  payload: {
    data: Course;
  };
}

interface CourseDetailLoading {
  type: typeof CourseTypes.CourseDetailLoading;
  payload: {
    loading: boolean;
  };
}

interface ListCourse {
  type: typeof CourseTypes.ListCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
    };
  };
}

interface ListCourseLoading {
  type: typeof CourseTypes.ListCourseLoading;
  payload: {
    loading: boolean;
  };
}

interface AddCourse {
  type: typeof CourseTypes.AddCourse;
  payload: {
    data: boolean;
  };
}

interface UpdateCourse {
  type: typeof CourseTypes.UpdateCourse;
  payload: {
    data: boolean;
  };
}

interface SubmitLoading {
  type: typeof CourseTypes.SubmitLoading;
  payload: {
    loading: boolean;
  };
}

interface ListCourseOwner {
  type: typeof CourseTypes.ListCourseOwner;
  payload: {
    data: CourseOwnerType;
  };
}

interface CourseOwnerLoading {
  type: typeof CourseTypes.CourseOwnerLoading;
  payload: {
    loading: boolean;
  };
}

interface UserCourse {
  type: typeof CourseTypes.UserCourse;
  payload: {
    data: UserCourseType;
  };
}

interface UserCourseLoading {
  type: typeof CourseTypes.UserCourseLoading;
  payload: {
    loading: boolean;
  };
}

interface UserWhoPaidCourse {
  type: typeof CourseTypes.UserWhoPaidCourse;
  payload: {
    data: User[];
  };
}

interface UserWhoPaidCourseLoading {
  type: typeof CourseTypes.UserWhoPaidCourseLoading;
  payload: {
    loading: boolean;
  };
}

export type CourseActionTypes =
  | ListAcceptedCourse
  | ListDeclinedCourse
  | ListPendingCourse
  | Loading
  | Errors
  | NumberOfAcceptedCourse
  | NumberOfDeclinedCourse
  | NumberOfPendingCourse
  | ListCourseDetail
  | CourseDetailLoading
  | ListCourse
  | ListCourseLoading
  | AddCourse
  | UpdateCourse
  | SubmitLoading
  | ListCourseOwner
  | CourseOwnerLoading
  | UserCourse
  | UserCourseLoading
  | UserWhoPaidCourse
  | UserWhoPaidCourseLoading;
