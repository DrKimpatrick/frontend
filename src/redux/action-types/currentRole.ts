export const CURRENT_ROLE_LOADING = 'CURRENT_ROLE_LOADING';
export const CURRENT_ROLE_FAIL = 'CURRENT_ROLE_FAIL';
export const CURRENT_ROLE_SUCCESS = 'FORGOT_SUCCESS';

export interface CurrentRole {
    email: string
}

export interface currentRoleLoading  {
    type: typeof CURRENT_ROLE_LOADING
}

export interface currentRoleFail  {
    type: typeof CURRENT_ROLE_FAIL,
    payload: string
}

export interface currentRoleSuccess  {
    type: typeof CURRENT_ROLE_SUCCESS,
    payload: any
}

export type CurrentRoleDispatchTypes = currentRoleLoading | currentRoleFail | currentRoleSuccess;
