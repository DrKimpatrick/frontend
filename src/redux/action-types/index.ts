// This is where we will add action types