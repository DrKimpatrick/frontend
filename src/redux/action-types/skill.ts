export enum SkillTypes {
  SkillLoading = 'SkillLoading/Loading',
  SkillSuccess = 'Skill/SkillSuccess',
  SkillFail = 'Skill/SkillFail',
  GetSkillsLoading = 'Skill/Loading',
  GetSkillsSuccess = 'Skill/GetSkillsSuccess',
  GetSkillsFail = 'Skill/GetSkillsFail',
  ListUserSkill = 'Skill/ListUserSkill',
  Errors = 'Skill/Errors',
  Loading = 'Skill/LoadingSkill',
  ChangeSkillStatus = 'Skill/ChangeSkillStatus',
  AddSkill = 'Skill/AddSkill',
  ListSingleSkill = 'Skill/ListSingleSkill',
  ListSkill = 'Skill/ListSkill',
  UpdateSkill = 'Skill/UpdateSkill',
  DeleteSkill = 'Skill/DeleteSkill',
  SetActionModal = 'Skill/SetActionModal',
  AddUserSkill = 'Skill/AddUserSkill',
  LoadingBtn = 'Skill/LoadingBtn'
}

export enum VerificationStatus {
  Unverified = 'unverified',
  InProgress = 'inProgress',
  Verified = 'verified'
}

export interface Skill {
  _id: string;
  skill: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface UserSkill {
  _id: string;
  skill: Skill;
  createdAt?: string;
  updatedAt: string;
  level: string;
  verificationStatus: VerificationStatus;
}
export interface AddedSkill {
  _id: string;
  skill: string;
  user: string;
  level: string;
  verficationLevel: VerificationStatus;
}

export interface AddSkill {
  skill: string;
  level: string;
  verificationStatus: VerificationStatus;
}

export enum SkillLevel {
  Beginner = 'beginner',
  Advanced = 'advanced',
  Intermediate = 'intermediate'
}

export interface AddSkillLoading {
  type: SkillTypes.SkillLoading;
}

export interface AddSkillFail {
  type: SkillTypes.SkillFail;
  payload: string;
}

export interface AddSkillSuccess {
  type: SkillTypes.SkillSuccess;
  payload: AddedSkill;
}

export interface GetSkillsLoading {
  type: SkillTypes.GetSkillsLoading;
}

export interface GetSkillsFail {
  type: SkillTypes.GetSkillsFail;
  payload: string;
}

export interface GetSkillsSuccess {
  type: SkillTypes.GetSkillsSuccess;
  payload: Skill;
}

interface ListUserSkill {
  type: typeof SkillTypes.ListUserSkill;
  payload: {
    data: UserSkill[];
  };
}
interface Errors {
  type: typeof SkillTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof SkillTypes.Loading;
  payload: {
    loading: boolean;
  };
}

interface ChangeSkillStatus {
  type: typeof SkillTypes.ChangeSkillStatus;
  payload: {
    data: UserSkill;
  };
}

interface AddSkillAsAdmin {
  type: typeof SkillTypes.AddSkill;
  payload: {
    data: Skill[];
  };
}

interface DeleteSkill {
  type: typeof SkillTypes.DeleteSkill;
  payload: {
    data: string;
  };
}

interface UpdateSkill {
  type: typeof SkillTypes.UpdateSkill;
  payload: {
    data: Skill;
  };
}

interface ListSkill {
  type: typeof SkillTypes.ListSkill;
  payload: {
    data: Skill[];
  };
}

interface ListSingleSkill {
  type: typeof SkillTypes.ListSingleSkill;
  payload: {
    data: Skill;
  };
}

interface SetActionModal {
  type: typeof SkillTypes.SetActionModal;
  payload: {
    data: boolean;
  };
}

interface AddUserSkill {
  type: typeof SkillTypes.AddUserSkill;
  payload: {
    data: boolean;
  };
}

interface LoadingBtn {
  type: typeof SkillTypes.LoadingBtn;
  payload: {
    loading: boolean;
  };
}

export type SkillActionTypes =
  | AddSkillLoading
  | AddSkillSuccess
  | AddSkillFail
  | GetSkillsFail
  | GetSkillsSuccess
  | GetSkillsLoading
  | ListUserSkill
  | Errors
  | Loading
  | ChangeSkillStatus
  | AddSkillAsAdmin
  | ListSingleSkill
  | UpdateSkill
  | DeleteSkill
  | ListSkill
  | SetActionModal
  | AddUserSkill
  | LoadingBtn;
