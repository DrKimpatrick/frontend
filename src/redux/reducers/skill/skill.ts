import {
  SkillTypes,
  SkillActionTypes,
  UserSkill,
  Skill
} from 'redux/action-types/skill';

interface InitialState {
  loading?: boolean;
  userSkill?: UserSkill[];
  errors?: any;
  skills?: Skill[];
  singleSkill?: Skill;
  action?: boolean;
  userSkillAction?: boolean;
  loadingBtn?: boolean;
}

const initialState: InitialState = {};

export const skillReducer = (
  state = initialState,
  action: SkillActionTypes
) => {
  switch (action.type) {
    case SkillTypes.ListUserSkill:
      return {
        ...state,
        userSkill: action.payload.data,
        loading: false,
        userSkillAction: false
      };

    case SkillTypes.Loading:
      return { ...state, loading: action.payload.loading };

    case SkillTypes.Errors:
      return { ...state, errors: action.payload.errors, loading: false };

    case SkillTypes.AddSkill:
      return {
        ...state,
        skills: state.skills
          ? state.skills.concat(action.payload.data)
          : action.payload.data,
        action: false
      };

    case SkillTypes.DeleteSkill:
      return {
        ...state,
        skills: state.skills
          ? state.skills.filter(item => item._id !== action.payload.data)
          : state.skills,
        action: false
      };

    case SkillTypes.UpdateSkill:
      if (state.skills) {
        const allSkills = state.skills;

        const skillIndex = allSkills.findIndex(
          item => item._id === action.payload.data._id
        );

        allSkills[skillIndex] = action.payload.data;

        return { ...state, skills: allSkills };
      }
      return state;

    case SkillTypes.ListSingleSkill:
      return { ...state, singleSkill: action.payload.data, loading: false };

    case SkillTypes.ListSkill:
      return {
        ...state,
        skills: action.payload.data,
        loading: false
      };

    case SkillTypes.SetActionModal:
      return { ...state, action: action.payload.data };

    case SkillTypes.AddUserSkill:
      return { ...state, userSkillAction: action.payload.data };

    case SkillTypes.LoadingBtn:
      return { ...state, loadingBtn: action.payload.loading };

    default:
      return state;
  }
};
