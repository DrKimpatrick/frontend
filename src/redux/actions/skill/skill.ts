import { Dispatch } from 'redux';
import { SkillTypes } from 'redux/action-types/skill';
import { UserTypes } from 'redux/action-types/user';
import { setMessage, setError } from 'redux/actions/message';
import ApiAction from '../../../helpers/apiAction';

export const listUserSkill = () => (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/skills/me',
      method: 'GET',
      onStart: () => (dispatch: Dispatch) => {
        dispatch({ type: SkillTypes.Loading, payload: { loading: true } });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.ListUserSkill,
          payload: { data: res.data }
        });
        dispatch({ type: SkillTypes.Loading, payload: { loading: false } });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: { errors: error.response.data }
        });
        dispatch({ type: SkillTypes.Loading, payload: { loading: false } });
      }
    })
  );
};

export const changeSkillStatus = (values: { status: string; id: string }) => (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/skills/status/${values.id}`,
      method: 'PUT',
      data: { verificationStatus: values.status },
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.ChangeSkillStatus,
          payload: { data: res.data }
        });

        setMessage(res.message)(dispatch);
      }
    })
  );
};

export const setSkillAction = (value: boolean) => (dispatch: Dispatch) => {
  dispatch({ type: SkillTypes.SetActionModal, payload: { data: value } });
};

export const listSkill = () => async (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/skills',
      method: 'GET',
      onStart: () => dispatch => {
        dispatch({
          type: SkillTypes.Loading,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.ListSkill,
          payload: { data: res.data }
        });
      }
    })
  );
};

export const listSingleSkill = (skillId: string) => async (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/skills/${skillId}`,
      method: 'GET',
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.ListSingleSkill,
          payload: { data: res.data }
        });
      }
    })
  );
};

export const deleteSkill = (skillId: string) => async (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/skills/${skillId}`,
      method: 'DELETE',
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
        setSkillAction(true)(dispatch);
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: () => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.DeleteSkill,
          payload: { data: skillId }
        });
        setSkillAction(false)(dispatch);
      }
    })
  );
};

export const updateSkill = ({
  skillId,
  skill
}: {
  skillId: string;
  skill: string;
}) => async (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: `/skills/${skillId}`,
      method: 'PUT',
      data: {
        skill
      },
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
        setSkillAction(true)(dispatch);
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.UpdateSkill,
          payload: { data: res.data }
        });

        setSkillAction(false)(dispatch);
      }
    })
  );
};

export const addSkill = (data: { skill: string }[]) => async (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: '/skills',
      method: 'POST',
      data,
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserSkillLoading,
          payload: { loading: true }
        });
        setSkillAction(true)(dispatch);
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.AddSkill,
          payload: { data: res.data }
        });

        setSkillAction(false)(dispatch);
      }
    })
  );
};

export const addUserSkill = (
  data: { skill: string; level: string }[]
) => async (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/skills/me',
      method: 'POST',
      data,
      onStart: () => dispatch => {
        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: false }
        });
      },
      onSuccess: () => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.AddUserSkill,
          payload: { data: true }
        });

        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: false }
        });
        listUserSkill()(dispatch);

        setMessage('skill saved successfully')(dispatch);
      }
    })
  );
};

export const updateUserSkill = (
  data: { userSkill: string; level: string }[]
) => async (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/skills/me',
      method: 'PATCH',
      data,
      onStart: () => dispatch => {
        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: false }
        });
      },
      onSuccess: () => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.AddUserSkill,
          payload: { data: true }
        });

        dispatch({
          type: SkillTypes.LoadingBtn,
          payload: { loading: false }
        });
        listUserSkill()(dispatch);

        setMessage('skill updated successfully')(dispatch);
      }
    })
  );
};

export const deleteUserSkill = (value: string[]) => async (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: '/skills/me',
      method: 'DELETE',
      data: value,
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
        setError('Failed to delete skill')(dispatch);
      },
      onSuccess: () => (dispatch: Dispatch) => {
        listUserSkill()(dispatch);

        setMessage('deleted successfully')(dispatch);
      }
    })
  );
};
