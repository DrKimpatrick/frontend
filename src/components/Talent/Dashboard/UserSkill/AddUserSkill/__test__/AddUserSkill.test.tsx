import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import { cleanup, render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import ReactDom from 'react-dom';
import { Store } from 'redux';
import thunk from 'redux-thunk';
import { VerificationStatus, SkillLevel } from 'redux/action-types/skill';
import Component from '../AddUserSkill';

let store: Store;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

const skillState = {
  skill: {
    skill: 'php',
    createdAt: new Date().toDateString(),
    updatedAt: new Date().toDateString()
  },
  _id: 'id',
  level: SkillLevel.Beginner,
  verificationStatus: VerificationStatus.Unverified,
  createdAt: new Date().toDateString(),
  updatedAt: new Date().toDateString()
};

const initialState = {
  skills: {
    userSkill: [skillState],
    skills: [skillState.skill, skillState.skill],
    loading: false,
    userSkillAction: false,
    loadingBtn: false,
    errors: {
      error: 'token',
      message: 'message'
    }
  }
};

const props = {
  initialValue: {
    skillName: 'skillName',
    skill: '',
    level: ''
  },
  close: jest.fn(),
  buttonName: 'save',
  heading: 'Add Skill',
  submit: jest.fn()
};

describe('AddUserSkill', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDom.render(
      <Provider store={store}>
        <Router>
          <Component {...props} />
        </Router>
      </Provider>,
      div
    );
  });

  it('should create snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <Component {...props} />
        </Router>
      </Provider>
    );

    expect(tree).toMatchSnapshot();
  });

  describe('Not Found', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        skills: { ...initialState.skills, userSkillAction: false, skills: [] }
      });
    });

    it('return not found message', () => {
      const tree = render(
        <Provider store={store}>
          <Router>
            <Component {...props} />
          </Router>
        </Provider>
      );

      expect(tree).toBeTruthy();
    });
  });

  describe('set user skillAction', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        skills: { ...initialState.skills, userSkillAction: true }
      });
    });

    it('set userSkillAction to true', () => {
      const tree = render(
        <Provider store={store}>
          <Router>
            <Component {...props} />
          </Router>
        </Provider>
      );

      expect(tree).toBeTruthy();
    });
  });
});
