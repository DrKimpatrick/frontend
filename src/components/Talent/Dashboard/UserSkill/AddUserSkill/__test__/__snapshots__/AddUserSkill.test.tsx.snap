// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserSkill should create snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 0px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
      role="presentation"
      style="position: fixed; z-index: 100; right: 0px; bottom: 0px; top: 0px; left: 0px;"
      tabindex="0"
    >
      <div
        aria-hidden="true"
        style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="addUserSkill"
        tabindex="-1"
      >
        <div
          class="title flex content-center"
        >
          <div
            class="backIcon"
            data-testid="goBack"
            style="cursor: pointer;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-backIcon-1"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </div>
          <div
            class="heading"
          >
            Add Skill
          </div>
        </div>
        <form
          autocomplete="off"
        >
          <small
            class="inputError"
          >
            token
          </small>
          <small
            class="inputError"
          >
            message
          </small>
          <div
            class="formGroup"
          >
            <div
              class="customSelect"
            >
              <select
                name="level"
              >
                <option
                  value=""
                >
                  select level
                </option>
                <option
                  value="beginner"
                >
                  beginner
                </option>
                <option
                  value="advanced"
                >
                  advanced
                </option>
                <option
                  value="intermediate"
                >
                  intermediate
                </option>
              </select>
              <span
                class="spanArrow"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="formGroup"
          >
            <div
              class="w-full flex content-center justify-center"
            >
              <button
                class="submitBtn"
                type="submit"
              >
                save
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
    <div
      aria-hidden="true"
    />
    <div
      role="presentation"
      style="position: fixed; z-index: 100; right: 0px; bottom: 0px; top: 0px; left: 0px;"
      tabindex="0"
    >
      <div
        aria-hidden="true"
        style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="addUserSkill"
        tabindex="-1"
      >
        <div
          class="title flex content-center"
        >
          <div
            class="backIcon"
            data-testid="goBack"
            style="cursor: pointer;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-backIcon-1"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </div>
          <div
            class="heading"
          >
            Add Skill
          </div>
        </div>
        <form
          autocomplete="off"
        >
          <small
            class="inputError"
          >
            token
          </small>
          <small
            class="inputError"
          >
            message
          </small>
          <div
            class="formGroup"
          >
            <div
              class="customSelect"
            >
              <select
                name="level"
              >
                <option
                  value=""
                >
                  select level
                </option>
                <option
                  value="beginner"
                >
                  beginner
                </option>
                <option
                  value="advanced"
                >
                  advanced
                </option>
                <option
                  value="intermediate"
                >
                  intermediate
                </option>
              </select>
              <span
                class="spanArrow"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="formGroup"
          >
            <div
              class="w-full flex content-center justify-center"
            >
              <button
                class="submitBtn"
                type="submit"
              >
                save
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
